packages/cli/src/modules/insights/insights.service.ts

```diff
@@ -35,7 +41,15 @@ const shouldSkipMode: Record<WorkflowExecuteMode, boolean> = {
 
 @Service()
 export class InsightsService {
-	constructor(private readonly sharedWorkflowRepository: SharedWorkflowRepository) {}
+	constructor(
+		private readonly sharedWorkflowRepository: SharedWorkflowRepository,
+		private readonly insightsByPeriodRepository: InsightsByPeriodRepository,
+		private readonly insightsRawRepository: InsightsRawRepository,
+	) {
+		// TODO: check if there is a better way to schedule this
+		const intervalMilliseconds = config.compactionIntervalMinutes * 60 * 1000;
+		setInterval(async () => await this.compactInsights(), intervalMilliseconds);
```

Comment:
Do we need a corresponding `clearInterval` for this?

Reply:
done,thanks !